### PR TITLE
LPS-151909 As a developer using headless endpoints, I'd like to update a country by countryId via partial update and full replace

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-151909](https://issues.liferay.com/browse/LPS-151909).<br/><br/>/cc @pei-jung @erickmonteiroo @ChrisKian @patriciajeanperez @epark18 @david-truong

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:forward

**Comment by drewbrokke** [comment]
ci:forward

File: portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java [hunk 1]
```diff
@@ -65,29 +66,9 @@ public Country addCountry(
...
-		if (fetchCountryByNumber(serviceContext.getCompanyId(), number) !=
-				null) {
-
-			throw new DuplicateCountryException(
-				"Number belongs to another country");
-		}
-
-		validate(a2, a3, idd, name, number);
+		_validate(
+			CountryConstants.DEFAULT_COUNTRY_ID, serviceContext.getCompanyId(),
```
Comment (pei-jung) [inline portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java:70]: Just realized that `CountryConstants.DEFAULT_COUNTRY_ID` points to [sql.data.com.liferay.portal.kernel.model.Country.country.id](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/portal.properties#L1760), which is not relevant anymore.  We should probably use an arbitrary number, like -1, to signal adding a new country.

**Comment by Sharryxue** [comment]
The [rest-builder-jdk8/0/0](https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/3528//consoleText) issue seems related to the commits, would you @drewbrokke please help with confirming?
```
     [echo] diff --git a/modules/apps/headless/headless-admin-address/headless-admin-address-impl/src/main/java/com/liferay/headless/admin/address/internal/resource/v1_0/BaseCountryResourceImpl.java b/modules/apps/headless/headless-admin-address/headless-admin-address-impl/src/main/java/com/liferay/headless/admin/address/internal/resource/v1_0/BaseCountryResourceImpl.java
     [echo] index 9b258db16a3c..aa70bdc27cb9 100644
     [echo] --- a/modules/apps/headless/headless-admin-address/headless-admin-address-impl/src/main/java/com/liferay/headless/admin/address/internal/resource/v1_0/BaseCountryResourceImpl.java
     [echo] +++ b/modules/apps/headless/headless-admin-address/headless-admin-address-impl/src/main/java/com/liferay/headless/admin/address/internal/resource/v1_0/BaseCountryResourceImpl.java
     [echo] @@ -664,12 +664,39 @@ public abstract class BaseCountryResourceImpl
     [echo]  			Map<String, Serializable> parameters)
     [echo]  		throws Exception {
     [echo]  
     [echo] -		for (Country country : countries) {
```
No others related failure found.
Thanks.

File: portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java (hunk 1 above)
Comment (drewbrokke) [reply portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java:70]: That's a great catch, Pei. I'll change that.

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
@epark18 are these failures related?

**Comment by epark18** [comment]
Failures are unrelated to this PR.

**Comment by epark18** [comment]
Pushed to Brian. https://github.com/brianchandotcom/liferay-portal/pull/117357

**Pull request closed without merging**